config/config_test.go

```diff
@@ -324,6 +435,13 @@ func TestLoadConfigFileWithGroups(t *testing.T) {
 	assert.Equal(t, expected, conf)
 }
 
+func TestLoadConfigFileWithMerge(t *testing.T) {
+	conf, err := parseCreationRuleForFile(parseConfigFile(sampleConfigWithMergeType, t), "/conf/path", "whatever", nil)
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(conf.KeyGroups[0]))
```

Comment:
I'd also add a test for the length of `conf.Keygroups` itself:
```suggestion
	assert.Equal(t, 2, len(conf.KeyGroups))
	assert.Equal(t, 1, len(conf.KeyGroups[0]))
```